include/onnxruntime/core/framework/execution_provider.h

```diff
@@ -99,6 +99,9 @@ class IExecutionProvider {
      * The return value is non-null if and only if a matching kernel was found.
      */
     virtual const KernelCreateInfo* LookUpKernel(const Node& node) const = 0;
+
+   protected:
+    ~IKernelLookup() = default;
```

Comment:
virtual

@edgchen1 , you missed the virtual keyword.

Reply:
hm, I think this is following the guideline:

"A base class destructor should be either public and virtual, or _protected and non-virtual_"

Reply:
This is strange. Would you be able to create a minimal reproducible example like what are in https://learn.microsoft.com/en-us/cpp/code-quality/c26436?view=msvc-170 ? Then you can create a VC++ bug with it, soon we will get a response. 

![image](https://user-images.githubusercontent.com/856316/192415166-16a937dc-379d-455c-8eb3-cd6f73685bf2.png)

Reply:
I think I need to also add `final` to the derived classes to ensure they are not also base classes. Will create a PR.